src/App.tsx

```diff
@@ -1,142 +1,298 @@
+
+  const [emptyFieldError, setEmptyError] = useState(false);
+  const [failedAddError, setAddError] = useState(false);
+  const [failedDeleteError, setDeleteError] = useState(false);
+  const [failedLoadError, setLoadError] = useState(false);
+  const [failedChangeError, setChangeError] = useState(false);
+
+  const cancelErrors = () => {
+    setEmptyError(false);
+    setAddError(false);
+    setDeleteError(false);
+    setLoadError(false);
+    setChangeError(false);
+  };
+
+  const fetchTodos = async () => {
+    try {
+      const loadedTodos = await getTodos(user?.id);
+
+      setTodoList(loadedTodos);
+      setVisibleList(loadedTodos);
+    } catch {
+      setLoadError(true);
+      setTimeout(cancelErrors, 3000);
+    }
+  };
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+
+    fetchTodos();
   }, []);
 
-  return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
+  const findNewTodoId = (): number => {
+    if (todoList.length !== 0) {
+      const max = todoList.reduce((a, b) => (a.id > b.id ? a : b));
 
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
+      return max.id + 1;
+    }
 
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
+    return 3456;
+  };
 
-        <section className="todoapp__main" data-cy="TodoList">
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
+  const pushTodos = async () => {
+    const pushedTodo = await pushTodo(findNewTodoId(), newTitle, user?.id);
 
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
-            <button
-              type="button"
-              className="todo__remove"
-              data-cy="TodoDeleteButton"
-            >
-              ×
-            </button>
+    setVisibleList(prevTodos => {
+      return [...prevTodos, pushedTodo];
+    });
+    setTodoList(prevTodos => {
+      return [...prevTodos, pushedTodo];
+    });
+  };
 
-            <div data-cy="TodoLoader" className="modal overlay">
-              <div className="modal-background has-background-white-ter" />
-              <div className="loader" />
-            </div>
-          </div>
+  const deleteTodos = async (id: number) => {
```

Comment:
why `Todos` - you delete 1 todo anyway. To avoid variable names duplication use word "remove" etc